data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfiguration.java

```diff
@@ -174,8 +175,22 @@ public static ConnectionConfiguration readConnectionConfiguration(final PluginSe
       builder = builder.withConnectTimeout(connectTimeout);
     }
 
-    builder.withAwsSigv4(pluginSetting.getBooleanOrDefault(AWS_SIGV4, false));
-    if (builder.awsSigv4) {
+    Map<String, Object> awsOption = pluginSetting.getTypedMap(AWS_OPTION, String.class, Object.class);
+    boolean awsOptionUsed = false;
+    builder.withAwsSigv4(false);
+    if (awsOption != null && !awsOption.isEmpty()) {
+        awsOptionUsed = true;
+        builder.withAwsSigv4(true);
```

Comment:
I rather like this actually - by having the `aws` properties we automatically use SigV4.